lib/src/actions/estimate_gas.dart

```diff
@@ -1,48 +1,101 @@
 import 'dart:js_interop';
 
+import 'package:freezed_annotation/freezed_annotation.dart';
 import 'package:wagmi_flutter_web/src/js/wagmi.js.dart';
+import 'package:wagmi_flutter_web/src/models/fees_values.dart';
 
-/// [Documentation API](https://wagmi.sh/core/api/actions/estimateGas)
-class EstimateGasParameters {
-  EstimateGasParameters({
-    this.accessList,
-    this.account,
-    this.chainId,
-    this.data,
-    this.gas,
-    this.gasPrice,
-    this.maxFeePerGas,
-    this.maxPriorityFeePerGas,
-    this.nonce,
-    this.to,
-    this.type,
-    this.value,
-  });
-  List<Map<String, dynamic>>? accessList;
-  String? account;
-  int? chainId;
-  String? data;
-  BigInt? gas;
-  BigInt? gasPrice;
-  BigInt? maxFeePerGas;
-  BigInt? maxPriorityFeePerGas;
-  int? nonce;
-  String? to;
-  String? type;
-  BigInt? value;
+part 'estimate_gas.freezed.dart';
+part 'estimate_gas.g.dart';
 
-  JSEstimateGasParameters get toJS => JSEstimateGasParameters(
-        accessList: accessList?.jsify() as JSArray<JSObject>?,
-        account: account?.toJS,
-        chainId: chainId?.toJS,
-        data: data?.toJS,
-        gas: gas?.toJS ?? BigInt.zero.toJS,
-        gasPrice: gasPrice?.toJS ?? BigInt.zero.toJS,
-        maxFeePerGas: maxFeePerGas?.toJS ?? BigInt.zero.toJS,
-        maxPriorityFeePerGas: maxPriorityFeePerGas?.toJS ?? BigInt.zero.toJS,
-        nonce: nonce?.toJS,
-        to: to?.toJS,
-        type: type?.toJS,
-        value: value?.toJS,
-      );
+@freezed
+class EstimateGasParameters with _$EstimateGasParameters {
+  const factory EstimateGasParameters({
```

Comment:
This could be an union class. It would explicitely show dart developers which attributes are expected.

Using Union classes, you wouldn't need for the `switch` statement in `toJS` method : it would be replaced by an extension on each Union class.